gateway/node.go

```diff
@@ -328,3 +328,7 @@ func (gw *Node) limit(ctx context.Context, tokens int) error {
 	}
 	return nil
 }
+
+func (gw *Node) EthGetBlockReceipts(ctx context.Context, blkParam ethtypes.EthBlockNumberOrHash) ([]*api.EthTxReceipt, error) {
+	return gw.target.EthGetBlockReceipts(ctx, blkParam)
+}
```

Comment:
put this in proxy_eth.go

also, see in there we have `EthGetTransactionReceipt`, we need to add a rate limit at the top checking `gw.limit(ctx, stateRateLimitTokens)`

and also, you can see in there why `EthGetTransactionReceiptLimited` exists - to pass the `maxMessageLookbackEpochs` through to the function. You should be able to follow that same pattern and add a `EthGetBlockReceiptsLimited`:

* add in api/api_full.go
* add in api/api_gateway.go
* add in the top of gateway/node.go
* add in gateway/proxy_eth.go along with `EthGetBlockReceipts`
* in node/impl/full/eth.go rename your current `EthGetBlockReceipts` to `EthGetBlockReceiptsLimited` with a `api.LookbackNoLimit` limit like `EthGetTransactionByHash` does

Reply:
The gateway/proxy* methods exist to impose limits for anyone exposing these APIs to the public, like glif etc. so you can impose rate limiting and also limitations on how far back in state you can look. I think glif does 1 day by default.

Reply:
Could you please explain a bit more about the limit? Currently, I am passing `abi.ChainEpoch` as the limit to the function. Is that correct or should it be something else, for tests, should I pass the limit as 1 or 0 to retrieve all the receipts or should it be something else?

Reply:
`gw.limit()` is a rate limiter based on the complexity of the query:

```
	basicRateLimitTokens  = 1
	walletRateLimitTokens = 1
	chainRateLimitTokens  = 2
	stateRateLimitTokens  = 3
```

the APIs that need to go digging around in state, (more than just basic chain queries), consume the most tokens; this API would be one of those, so just use `gw.limit(ctx, stateRateLimitTokens)` to "burn up" 3 tokens for that query. The internals of the rate limiter have a mechanism to account for those and either slow a user down or block them if they consume too many of these tokens in a short period.

For the `limit` argument to `EthGetBlockReceiptsLimited` from `EthGetBlockReceipts`, just use `api.LookbackNoLimit` and then copy the logic you see in `EthGetTransactionReceiptLimited` from proxy_eth.go - where it gets capped at `gw.maxMessageLookbackEpochs`. What that does is lets the gateway user run it with `--api-max-lookback` and a number of epochs, such as `2880`. So when you end up resolving the epoch that the `EthGetBlockReceiptsLimited` call (in eth.go), you can compare that to the `limit` parameter that gets passed in and reject it if the block is too far in the past.